docs/sop/MIGRATION_TROUBLESHOOT.md

```diff
@@ -0,0 +1,105 @@
+works with beta.gigadb.org, gigadb.org and portainer.gigadb.org domains. The
+procedure of how this TLS certificate was created is documented below.
+
+Log into live web ec2 instance:
+```
+$ cd ops/infrastructure/envs/live
+$ ssh -i ~/.ssh/id-rsa-aws-hk-gigadb.pem -o ProxyCommand="ssh -W %h:%p -i ~/.ssh/id-rsa-aws-hk-gigadb.pem centos@ec2_bastion_public_ip" centos@ec2_private_ip
+```
+
+When you list available docker images, you will see a certbot docker image:
+```
+[centos@ip-10-99-0-229 ~]$ docker images
+REPOSITORY                                                                           TAG       IMAGE ID       CREATED         SIZE
+registry.gitlab.com/gigascience/upstream/gigadb-website/production_app               live      5471da52eb3e   13 hours ago    611MB
+registry.gitlab.com/gigascience/upstream/gigadb-website/production_web               live      2e84a1281100   13 hours ago    111MB
+portainer/portainer-ce                                                               latest    7536f6a42542   5 months ago    294MB
+certbot/certbot                                                                      latest    28ba94451daa   5 months ago    115MB
+registry.gitlab.com/gigascience/upstream/gigadb-website/production_tideways-daemon   live      c51e5a87bce7   11 months ago   116MB
+registry.gitlab.com/gigascience/upstream/gigadb-website/production_config            live      669e50b0e226   22 months ago   15.3MB
+```
+
+Certbot is a free, open source tool for creating Let's Encrypt certificates that
+can be used by websites to enable HTTPS. We can create and run a new container
+using certbot docker image:
+```
+[centos@ip-10-99-0-229 ~]$ docker run -it certbot/certbot:latest --version
+certbot 2.7.4
+```
+
+The keys and certificates created by Certbot can be found in `/etc/letsencrypt/live$domain`.
+We need to mount this Docker container directory with a directory on the host
+machine. Let's create 3 new directories for this:
+```
+[centos@ip-10-99-0-229 ~]$ mkdir -p /home/centos/data/certbot/letsencrypt
+[centos@ip-10-99-0-229 ~]$ mkdir -p /home/centos/data/certbot/www
+[centos@ip-10-99-0-229 ~]$ mkdir -p /home/centos/data/certbot/log
+```
+
+There are rate limits to the number of certificates that Let's Encrypt will
+allow you to create. Over-running this limit means you will need to wait for a
+week before you will be allowed to create more certificates. For this reason, 
+use the Let's Encrypt test staging API to learn how to create TLS certificates
+by using `--test-cert` flag:
+```
+[centos@ip-10-99-0-229 ~]$ docker run -it --rm --name temp_certbot \
+  -v /home/centos/data/certbot/letsencrypt:/etc/letsencrypt \
+  -v /home/centos/data/certbot/www:/tmp/letsencrypt \
+  -v /home/centos/data/certbot/log:/var/log \
+  certbot/certbot:latest \
+  certonly --test-cert --manual \
+  --preferred-challenges dns \
+  -d gigadb.org -d beta.gigadb.org -d portainer.gigadb.org \
+  --email tech@gigasciencejournal.com
+```
```

Comment:
Hi @pli888 

When I executed this cmd in my staging webapp server, I would get this error:
```
[centos@ip-10-99-0-250 ~]$ docker run -it --rm --name temp_certbot -v /home/centos/data/certbot/letsencrypt:/etc/letsencrypt -v /home/centos/data/certbot/www:/tmp/letsencrypt -v /home/centos/data/certbot/log:/var/log certbot/certbot:latest certonly --test-cert --manual --preferred-challenges dns -d ec2-staging.gigadb.link -d portainer.ec2-staging.gigadb.link --email ken@gigasciencejournal.com
Certbot failed to authenticate some domains (authenticator: manual). The Certificate Authority reported these problems:
  Domain: ec2-staging.gigadb.link
  Type:   dns
  Detail: DNS problem: NXDOMAIN looking up TXT for _acme-challenge.ec2-staging.gigadb.link - check that a DNS record exists for this domain

  Domain: portainer.ec2-staging.gigadb.link
  Type:   dns
  Detail: DNS problem: NXDOMAIN looking up TXT for _acme-challenge.portainer.ec2-staging.gigadb.link - check that a DNS record exists for this domain

Hint: The Certificate Authority failed to verify the manually created DNS TXT records. Ensure that you created these in the correct location, or try waiting longer for DNS propagation on the next attempt.

Some challenges have failed.
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /var/log/letsencrypt/letsencrypt.log or re-run Certbot with -v for more details.
[centos@ip-10-99-0-250 ~]$ less /var/log/letsencrypt/letsencrypt.log
/var/log/letsencrypt/letsencrypt.log: No such file or directory
[centos@ip-10-99-0-250 ~]$ 
```

Reply:
Did you create the DNS TXT record for `_acme-challenge.ec2-staging.gigadb.link` using your DNS manager? If not then you will see this error because the DNS challenge failed. This is mentioned in line 84 onwards.

Reply:
After the creating the TXT record with prefix `_acme-challenge.ec2-staging` in the DNS console, I could execute the above cmd successfully.